source/extensions/filters/network/mysql_proxy/mysql_codec_clogin_resp.cc

```diff
@@ -227,103 +73,103 @@ DecodeStatus ClientLoginResponse::parseMessage(Buffer::Instance& buffer, uint32_
   }
   switch (resp_code) {
```

Comment:
What about this:
redesign _type_ to take resp_code as parameter, allocate proper type of the message and return a pointer to the message. The code here would look like this:
```
ClientLoginResponseMessage* message = allocMessageType(resp_code);
if (message != nullptr) {
return message->parseMessage(....
}
return DecodeStatus::Failure;

```

Reply:
We still need a method like `type(ClientLoginRespType)` to init inner `message` from zero to one but not from `parseMessage(buffer, len)`. I think to create a private method `void type(resp_code)` to init `message_` when `parseMessage`, and preserve the public method `void type(ClientLoginRespType)`, let the test functions init `message_` by passing `ClientLoginRespType`. What do you think?

Reply:
I do not exactly understand the point. Maybe you can put pseudo code here for better picture. Thanks.

Reply:
```cpp
class ClientLoginResponse{
void type(ClientLoingResponseType type);
private:
void type(uint8_t resp_code);
};

void parseMessage(Buffer::Instance& buffer, int len) {
//get respcode
type(resp_code); // init message_ by resp code
}

// mysql_clogin_resp_test.cc
void initOkMessage() {
ClientLoginResponse ok{};
ok.type(ClientLoginResponseType::Ok); // init message_ by ClientLoginResponseType
...
}
```
Or remove `type(ClientLoginResponseType)`, just use `type(uint8_t)` like below:
```cpp
ClientLoginResponse ok{};
ok.type(MYSQL_RESP_OK); // init message_ by resp_code
```